tests/test.nf.test

```diff
@@ -0,0 +1,93 @@
+nextflow_pipeline {
+
+    name "Test pipeline: NFCORE_EAGER"
+    script "main.nf"
+    tag "pipeline"
+    tag "nfcore_eager"
+    tag "test"
+
+    test("test_profile") {
+
+        when {
+            params {
+                outdir = "$outputDir"
+            }
+        }
+
+        then {
+
+            // Each top level results directory should be tested with individual snapshots/variables
+            // stable_name_<dir> is for files with variable md5sums (i.e. content) so only names will be compared
+            // stable_content_<dir> is for files with stable md5sums (i.e. content) so md5sums will be compared
+            // For both cases, use the third argument to specify globs to exclude certain files
+            // If a directory is fully stable, you can drop `stable_name_*`
+            // Generate with: nf-test test --tag test --profile docker,test --update-snapshot
+            // Test with:     nf-test test --tag test --profile docker,test
+            // NOTE: BAMs are always only stable in name, because:
+            //   a) sharding breaks header since the shard that was first is named in the header
+            //   b) the order of the reads in the BAMs is not stable (sorted, but reads that share a start position can be in any order)
+            //   point b) also causes BAIs to be unstable.
+
+            def stable_name_all                 = getAllFilesFromDir("$outputDir/"                  , true , ['pipeline_info/*']                 , null , ['*', '**/*'] )
```

Comment:
AS this is our main header exampel, can you leave a comment (aligned) with each seetionfo the getAllFilesFromDir bit)?

Although I realise now if we are dumping this test it's sort of not necessary anymore :sweat_smile: 

Unless we keep this as our default test anyway (Assuming it is)

And have our suggestions in #1103 as additional